neo4j/v1/session.py

```diff
@@ -143,60 +125,124 @@ class Result(list):
     #: Dictionary of parameters passed with the statement.
     parameters = None
 
-    def __init__(self, session, statement, parameters):
-        super(Result, self).__init__()
-        self.session = session
+    def __init__(self, connection, statement, parameters):
+        super(ResultCursor, self).__init__()
         self.statement = statement
         self.parameters = parameters
         self.keys = None
-        self.more = True
-        self.summary = None
-        self.bench_test = None
+        self._connection = connection
+        self._current = None
+        self._next = deque()
+        self._position = -1
+        self._summary = None
+        self._consumed = False
+
+    def is_open(self):
+        """ Return ``True`` if this cursor is still open, ``False`` otherwise.
+        """
+        return bool(self._connection)
 
-    def on_header(self, metadata):
-        """ Called on receipt of the result header.
+    def close(self):
+        """ Consume the remainder of this result and detach the connection
```

Comment:
To a novice user, not sure what "detach" means - what does that mean for how I write my code?